autoload/lsp/omni.vim

```diff
@@ -88,13 +88,22 @@ function! lsp#omni#complete(findstart, base) abort
 
             let s:completion['status'] = ''
 
+    		if s:should_skip() | return | endif
```

Comment:
Can we inline instead of creating a new function called `s:should_skip()`. 

Also I think you need to return some value instead of just using return.

Reply:
Actually we should move this check in `s:display_completions()` and don't call complete if not in insert mode and remove the try..catch.